tests/functionality/test_annotate_models.py

```diff
@@ -11,6 +15,65 @@
 from genmod.log import init_log
 init_log(logger, loglevel="INFO")
 
+def _parse_variant_file(file_path: str) -> HeaderParser:
+    """
+    Parse VCF header fields
+    :param file_path: VCF to be read
+    :raises ValueError: in case file is empty
+    """
+    with open(file_path, 'r') as variant_file:
+        head = HeaderParser()
+        for line_index, line in enumerate(variant_file):
+            line = line.rstrip()
+            if line.startswith('#'):
+                if line.startswith('##'):
+                    head.parse_meta_data(line)
+                else:
+                    head.parse_header_line(line)
+            else:
+                break
+        if line_index == 0:
+            raise ValueError('Expected contents in file, got none')
+    return head
+
+
+def _generate_variants_from_file(file_path: str) -> Dict[str, Union[str, int, float]]:
+    """
+    Yield variants from VCF file.
+    :param file_path: VCF to be read
+    """
+    header = _parse_variant_file(file_path=file_path)
+    with open(file_path, 'r') as variant_file:
+        for line in variant_file:
+            if line.startswith('#'):
+                continue
+            variant: Dict[str, str] = get_variant_dict(line, header.header)
+            variant['info_dict'] = get_info_dict(variant['INFO'])
+            yield variant
+
```

Comment:
One suggestion would be to move the variant generation methods into a fixture in the `functionality` module, so that we reduce duplicate code, related https://github.com/fellen31/genmod/commit/bc15a2f2e81f7b4f9f676dcdc460e6df192c68b9

Reply:
Thanks for the review @torbjorgen, I'll look into moving it into a fixture, and just ping you afterwards so you can make sure it looks ok.